src/zkp/generator.rs

```diff
@@ -1,7 +1,79 @@
 use core::{fmt, str};
 use ffi;
-use {constants, from_hex, Error, Secp256k1, SecretKey, Signing, Tag};
+#[cfg(feature = "rand")]
+use rand::Rng;
+use {constants, from_hex, Error, Secp256k1, Signing, Tag};
+/// Represents a blinding factor/Tweak on secp256k1 curve
+///
+/// This is same as [`crate::SecretKey`], but is allowed to take
+/// zero value and values outside of secp curve order
```

Comment:
Given that the definition of a `SecretKey` is that it cannot be zero and must be within the curve order, the only thing that this shares with a `SecretKey` is the length right?

What do you think about expressing this as:

```suggestion
/// Represents a blinding factor/Tweak on secp256k1 curve
///
/// Contrary to a [`crate::SecretKey`], any 32 bytes are a valid tweak. This includes `0` and values outside of the secp curve order.
```

Reply:
Shall we also add a `From<[u8; 32]>` implementation?